plugins/lyrics-genius/front.js

```diff
@@ -35,33 +49,58 @@ module.exports = () => {
 			return;
 		}
 
-		lyricsTab.removeAttribute("disabled");
-		lyricsTab.removeAttribute("aria-disabled");
-		document.querySelector("tp-yt-paper-tab").onclick = () => {
-			lyricsTab.removeAttribute("disabled");
-			lyricsTab.removeAttribute("aria-disabled");
-		};
+		enableLyricsTab();
+
+		setTabsOnclick(enableLyricsTab);
+
+		checkLyricsContainer();
 
-		lyricsTab.onclick = () => {
+		tabs.lyrics.onclick = () => {
 			const tabContainer = document.querySelector("ytmusic-tab-renderer");
 			const observer = new MutationObserver((_, observer) => {
-				const lyricsContainer = document.querySelector(
-					'[page-type="MUSIC_PAGE_TYPE_TRACK_LYRICS"] > ytmusic-message-renderer'
-				);
-				if (lyricsContainer) {
-					lyricsContainer.innerHTML = `<div id="contents" class="style-scope ytmusic-section-list-renderer genius-lyrics">
-			 			${lyrics}
-
-			 			<yt-formatted-string class="footer style-scope ytmusic-description-shelf-renderer">Source&nbsp;: Genius</yt-formatted-string>
-					</div>`;
-					observer.disconnect();
-				}
+				checkLyricsContainer(() => observer.disconnect());
 			});
 			observer.observe(tabContainer, {
 				attributes: true,
 				childList: true,
 				subtree: true,
 			});
 		};
+
+		function checkLyricsContainer(callback = () => {}) {
+			const lyricsContainer = document.querySelector(
+				'[page-type="MUSIC_PAGE_TYPE_TRACK_LYRICS"] > ytmusic-message-renderer'
+			);
+			if (lyricsContainer) {
+				callback();
+				setLyrics(lyricsContainer)
+			}
+		}
+
+		function setLyrics(lyricsContainer) {
+			lyricsContainer.innerHTML = 
+				`<div id="contents" class="style-scope ytmusic-section-list-renderer description ytmusic-description-shelf-renderer genius-lyrics">
+			 		${hasLyrics ? lyrics : 'Could not retrieve lyrics from genius'}
+
+				</div>
+				<yt-formatted-string class="footer style-scope ytmusic-description-shelf-renderer" style="align-self: baseline"></yt-formatted-string>`;
+			if (hasLyrics) {
+				lyricsContainer.querySelector('.footer').textContent = 'Source: Genius';
+				enableLyricsTab();
+			}
+		}
+
+		function setTabsOnclick(callback) {
+			for (tab of [tabs.upNext, tabs.discover]) {
```

Comment:
nit to avoid a warning about `tab`:
```suggestion
			for (const tab of [tabs.upNext, tabs.discover]) {
```